Use bundler

jherdman opened the pull request:
Update to use Bundler, and its various tools.

Commit:
Use Bundler

Commit:
Clean up gemspec a bit

Commit:
Update Rakefile for use with Bundler

Commit:
Move specs into canonical location

Commit:
RSpec helper file

Commit:
RSpec options

Commit:
Fix version handling

Commit:
Fix spec requiring

Commit:
rvmrc, because we like it tidy

dj2 commented on em-jack.gemspec:
```diff
@@ -1,28 +1,27 @@
+  s.required_rubygems_version = '>= 1.3.6'
 
-  s.add_dependency('eventmachine')
+  s.add_dependency 'eventmachine', ['>= 1.0.0.beta.3']
```
Jack should work with older versions of eventamchine. At a max, I think this should be 0.12.10 until 1.0 is out.

dj2 commented:
This is looking good. One thing, I'm not a fan of having .rvmrc files committed.  If you can remove that and bump the required EM version down this looks merge-able.

Commit:
Remove rvmrc, and ignore any that a dev might have

Commit:
Use older EventMachine for now

The usage of >= should allow projects using more recent versions to
coexist with em-jack.

jherdman commented:
Take a look at 8cadfff9733ee9b7f701846f034e8644c646848d. I think that should satisfy some projects I've encountered using the EventMachine betas, and those using older EventMachine.

dj2 commented:
Looks good, thanks for the fixes.
